src/utils/analytics.js

```diff
@@ -92,5 +116,28 @@ export function useAnalytics(params) {
     }
     locationRef.current = window?.location
   }, [])
-  return { analytics: data?.analytics, identify }
+  return { analytics: data?.analytics, identify, track }
+}
+
+export function useEmailCaptureTracking(defaultProperties) {
+  const { identify, track } = useAnalytics()
+
+  const handleSubmit = useCallback(
+    async ({ email, optInContext, optIn }) => {
+      const traits = {
+        Email: email,
+        "Email Capture Opt In": optInContext || undefined,
+        "Email Capture Opt In Checked": optInContext ? optIn : undefined,
+      }
+      await identify(traits)
+      await track("Marketing Site Form Submission", {
+        Type: "Email Capture",
+        ...defaultProperties,
+        ...traits,
+      })
```

Comment:
Do we want to wait for `identify` before calling `track`? Or abort the second call if the first one fails?
If not, we could make those calls independently:

```suggestion
      await Promise.all([
        identify(traits),
        track("Marketing Site Form Submission", {
          Type: "Email Capture",
          ...defaultProperties,
          ...traits,
        })
      ])
```

Reply:
Yes, we want `identify` before `track`, because we want to make sure the user is identified before tracking events for him

Reply:
Got it!